src/main/java/com/conveyal/datatools/manager/jobs/MergeFeedsJob.java

```diff
@@ -0,0 +1,584 @@
+                    boolean skipRecord = false;
+                    String[] rowValues = new String[specFields.size()];
+                    String[] values = csvReader.getValues();
+                    if (values.length == 1) {
+                        LOG.warn("Found blank line. Skipping...");
+                        continue;
+                    }
+                    // Piece together the row to write, which should look practically identical to the original
+                    // row except for the identifiers receiving a prefix to avoid ID conflicts.
+                    for (int specFieldIndex = 0; specFieldIndex < specFields.size(); specFieldIndex++) {
+                        Field field = specFields.get(specFieldIndex);
+                        // Get index of field from GTFS spec as it appears in feed
+                        int index = fieldsFoundList.indexOf(field);
+                        String val = csvReader.get(index);
+                        // Determine if field is a GTFS identifier.
+                        boolean isKeyField = field.isForeignReference() || keyField.equals(field.name);
+                        Set<NewGTFSError> idErrors = table.checkReferencesAndUniqueness(keyValue, lineNumber, field, val, referenceTracker, keyField, orderField);
+                        // Store values for key fields that have been encountered.
+                        switch (table.name) {
+                            // case "calendar_dates":
+                            case "calendar":
+                                // If any service_id in the active feed matches with the future feed, it should be
+                                // modified and all associated trip records must also be changed with the modified
+                                // service_id.
+                                // TODO How can we check that calendar_dates entries are duplicates? I think we would
+                                //  need to consider the service_id:exception_type:date as the unique key and include any
+                                //  all entries as long as they are unique on this key.
+                                for (NewGTFSError error : idErrors) {
+                                    if (error.errorType.equals(NewGTFSErrorType.DUPLICATE_ID)) {
+                                        String key = String.join(":", table.name, idScope, val);
+                                        // Modify service_id and ensure that referencing trips have service_id updated.
+                                        val = String.join(":", idScope, val);
+                                        mergeFeedsResult.remappedIds.put(key, val);
+                                    }
+                                }
+                                // If a service_id from the active calendar has both the start_date and end_date in the
+                                // future, the service will be excluded from the merged file. Records in trips,
```

Comment:
ok, but why? I don't understand why future services wouldn't be included in a merged feed.

Reply:
I think this might be ok after I realized what is happening down at https://github.com/conveyal/datatools-server/pull/186/files#r272345429. If so, mark this as resolved and let's discuss at that other comment.